yass-components-react/src/components/Box/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { ReactNode } from "react";
+
+import * as classNames from "../../../types/yass.json";
+
+export const tags = ["div", "span"];
+
+export interface BoxProps {
+  id?: string;
+  testId?: string;
+  children?: ReactNode;
+  as?: typeof tags[number];
+  display?: typeof classNames["display"][number];
+  position?: typeof classNames["position"][number];
+  padding?: typeof classNames["padding"][number];
+  paddingBlock?: typeof classNames["padding-block"][number];
+  paddingInline?: typeof classNames["padding-inline"][number];
+  backgroundColor?: typeof classNames["background-color"][number];
+  borderWidth?: typeof classNames["border-width"][number];
+  borderColor?: typeof classNames["border-color"][number];
+  borderStyle?: typeof classNames["border-style"][number];
+  borderRadius?: typeof classNames["border-radius"][number];
+  overflow?: typeof classNames["overflow"][number];
+  opacity?: typeof classNames["opacity"][number];
+  zIndex?: typeof classNames["z-index"][number];
+}
+
+const Box = ({
+  id,
+  testId,
+  as = "div",
+  padding,
+  paddingBlock,
+  paddingInline,
+  backgroundColor,
+  borderWidth,
+  borderColor,
+  borderStyle,
+  borderRadius,
+  display = "display:flex",
+  overflow,
+  opacity,
+  zIndex,
+  children,
+}: BoxProps) => {
+  const Component = as;
+
+  return (
+    // @ts-ignore TODO: figure out why
```

Comment:
`<Component as='some html tad name' ...` is giving me a type error (in `Text` as well). Does not affect how the code runs though, so I'll figure it out later.